src/main/java/address/updater/BackupHandler.java

```diff
@@ -152,14 +138,30 @@ public void cleanupBackups() {
             }
         });
 
-        dependencyHistoryHandler.cleanUpUnusedDependencies(unusedVersions);
+        dependencyHistoryHandler.cleanUpUnusedDependencies(deletedVersions);
+    }
+
+    private Version deleteBackupFile(String backupFileName) {
```

Comment:
Try to avoid method that has side effects and returns a value at the same time. It's called the [Command-Query separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation)